File: Sources/structlyC/03_beginner_recursion/sum_of_lengths.cpp
//
//  sum_of_lengths.h
//  MyCLI
//
//  Created by new on 3/14/26.
//

#include <vector>
#include <string>

int sumOfLengths_helper(int i, int acc, std::vector<std::string> strings) {
    if (i == strings.size()) {
        return acc;
    }
    return sumOfLengths_helper(i+1, acc + strings[i].size(), strings);
}

int sumOfLengths(std::vector<std::string> strings) {
  return sumOfLengths_helper(0, 0, strings);
}

File: Sources/structy/03_beginner_recursion/SumOfLengths.swift
//
//  SumOfLengths.swift
//  MyCLI
//
//  Created by new on 3/14/26.
//

func sum_of_lengths(_ strings: [String]) -> Int {
    func sol(i: Int, acc: Int, strings: [String]) -> Int {
        if i == strings.count {
            return acc
        }
        return sol(i: i+1, acc: acc+strings[i].count, strings: strings)
    }
    return sol(i: 0, acc: 0, strings: strings)
}
